src/arch.rs

```diff
@@ -159,6 +159,37 @@ pub trait Arch {
         None
     }
 
+    /// (optional) (LLDB extension) Return a `CallbackToken` that confirms that
+    /// the arch's register info of a single register has been send.
+    ///
+    /// Implementing this method enables LLDB to dynamically query the
+    /// target's register information one by one.
+
+    /// Some targets don't have register
+    /// context in the compiled version of the debugger. Help the debugger by
+    /// dynamically supplying the register info from the target. The debugger
+    /// will request the register info in a sequential manner till an error
+    /// packet is received. In LLDB, the register info search has the following [order](https://github.com/llvm/llvm-project/blob/369ce54bb302f209239b8ebc77ad824add9df089/lldb/source/Plugins/Process/gdb-remote/ProcessGDBRemote.cpp#L397-L402):<br/><br/>
+    ///     1.    Use the target definition python file if one is specified.  
+    ///     2.    If the target definition doesn't have any of the info from the
+    ///     target.xml (registers) then proceed to read the target.xml.   
+    ///     3.    Fall back on the `qRegisterInfo` packets.    
+    ///     4.    Use hardcoded defaults if available.   
+    ///
+    /// See the [LLDB gdb-remote docs](https://github.com/llvm-mirror/lldb/blob/d01083a850f577b85501a0902b52fd0930de72c7/docs/lldb-gdb-remote.txt#L396)
+    /// for more details on the available information that a single register can
+    /// be described by and [#99](https://github.com/daniel5151/gdbstub/issues/99) for more information on LLDB compatibility.
+
+    /// Invoke `write_info.write(reg)` where `reg` is the `Register` struct from
+    /// this module to write information of a single register or
+    /// `write_info.done()` if you want to end the `qRegisterInfo` packet
+    /// exchange. These two methods will return a `CallbackToken`, which has to
+    /// be returned from this function.
+    #[inline(always)]
+    fn register_info(_reg_id: usize, _write_info: Callback<'_>) -> Option<CallbackToken<'_>> {
```

Comment:
nit: avoid using `_name` in trait defns, as this leaks into the rustdoc for the signature. Instead, just add a `let _ = (reg_id, write_info);` within the function body to squelch the warning.